File: occupancy-manager-java/src/test/java/com/occupancy/manager/factory/OccupancyUsageResponseFactory.java
package com.occupancy.manager.factory;

import com.occupancy.manager.dto.OccupancyUsageResponse;
import com.occupancy.manager.util.JsonUtil;

/**
 * 
 * @author Ramachandra
 *
 */
public class OccupancyUsageResponseFactory {

	/**
	 * 
	 * @return OccupancyUsageResponse
	 */
	public static OccupancyUsageResponse createOccupancyUsageResponseFor2Premium7EconomyRooms() {
		return (OccupancyUsageResponse) new JsonUtil().readFileAndGetData("2Premium7EconomyRoomsResponse.json",
				OccupancyUsageResponse.class);
	}

	/**
	 * @return OccupancyUsageResponse
	 */
	public static OccupancyUsageResponse createOccupancyUsageResponseFor3Premium3EconomyRooms() {
		return (OccupancyUsageResponse) new JsonUtil().readFileAndGetData("3Premium3EconomyRoomsResponse.json",
				OccupancyUsageResponse.class);
	}

	/**
	 * @return OccupancyUsageResponse
	 */
	public static OccupancyUsageResponse createOccupancyUsageResponseFor7Premium1EconomyRoom() {
		return (OccupancyUsageResponse) new JsonUtil().readFileAndGetData("7Premium1EconomyRoomResponse.json",
				OccupancyUsageResponse.class);
	}

	/**
	 * @return OccupancyUsageResponse
	 */
	public static OccupancyUsageResponse createOccupancyUsageResponseFor7Premium5EconomyRooms() {
		return (OccupancyUsageResponse) new JsonUtil().readFileAndGetData("7Premium5EconomyRoomsResponse.json",
				OccupancyUsageResponse.class);
	}

}

File: occupancy-manager-kotlin/src/test/kotlin/com/occupancy/manager/factory/OccupancyUsageResponseFactory.kt
package com.occupancy.manager.factory

import com.occupancy.manager.dto.OccupancyUsageResponse
import com.occupancy.manager.util.JsonUtil

/**
 *
 * @author Ramachandra
 */
object OccupancyUsageResponseFactory {
	/**
	 *
	 * @return OccupancyUsageResponse
	 */
	fun createOccupancyUsageResponseFor2Premium7EconomyRooms(): OccupancyUsageResponse {
		return JsonUtil().readFileAndGetData(
			"2Premium7EconomyRoomsResponse.json",
			OccupancyUsageResponse::class.java
		) as OccupancyUsageResponse
	}

	/**
	 * @return OccupancyUsageResponse
	 */
	fun createOccupancyUsageResponseFor3Premium3EconomyRooms(): OccupancyUsageResponse {
		return JsonUtil().readFileAndGetData(
			"3Premium3EconomyRoomsResponse.json",
			OccupancyUsageResponse::class.java
		) as OccupancyUsageResponse
	}

	/**
	 * @return OccupancyUsageResponse
	 */
	fun createOccupancyUsageResponseFor7Premium1EconomyRoom(): OccupancyUsageResponse {
		return JsonUtil().readFileAndGetData(
			"7Premium1EconomyRoomResponse.json",
			OccupancyUsageResponse::class.java
		) as OccupancyUsageResponse
	}

	/**
	 * @return OccupancyUsageResponse
	 */
	fun createOccupancyUsageResponseFor7Premium5EconomyRooms(): OccupancyUsageResponse {
		return JsonUtil().readFileAndGetData(
			"7Premium5EconomyRoomsResponse.json",
			OccupancyUsageResponse::class.java
		) as OccupancyUsageResponse
	}
}